components/combobox/combobox.jsx

```diff
@@ -615,6 +623,16 @@ class Combobox extends React.Component {
 			options.push(...localProps.optionsSearchEntity);
 		}
 
+		if (props.hasDeselect) {
```

Comment:
Two observations:
1. instead of continuing the anti pattern requiring the eslint consider writing a method that returns an extended options list with the expected deselect object
2. this condition is also seen in the menu.jsx meaning there is opportunity to dry this up

If a new shared method is created then a single unit test against the shared method can be created.

Reply:
`options` array has been updated to not mutate